### DISCUSSION TITLE
URock for multiple timesteps

### DISCUSSION THREAD

**Discussion by bweeding** [UMEP-dev/UMEP#562] [Ideas]
Hi all, I'm looking into ways of running URock for multiple timesteps. In particular, I'm looking at ways this could be optimised ie. taking a met file in, identifying the unique combinations of wind speed and direction, running URock for those, and then outputting the results (I tend to work in xarray as much as possible). I also feel like there's an opportunity to set the URock code up with a particular wind direction, and then run it for multiple wind speeds to reduce the overheads of setting up every time. 

I'm working on this now and will share any major breakthroughs as I go, but I'd love to hear from anyone else who has tried this!

Cheers
Ben

**Comment by j3r3m1** [comment]
Hi @bweeding thanks for your message and sharing your ideas.

@simonmarti1992 and @bbrangeo you may have faced this question I suppose ?

Running URock with multiple wind speed for a given wind direction is indeed a good option since most of the initialization step is dedicated to the calculation of the wind factor for a given direction (and wind factors are independent from the wind speed). It means that for a given wind direction you should run the code up to [this line](https://github.com/UMEP-dev/UMEP-processing/blob/72f9eb6a62415d95f29b914862ed98821ccc9ae1/functions/URock/MainCalculation.py#L599) and then the rest of the code for each new wind speed.

**Comment by j3r3m1** [comment]
I suppose updating URock to account for a meteorological file as input could be a good improvement. Please feel free to make a pull request or sharing your proposition of modification on your own fork if you want to do that.

**Comment by s-martiz** [comment]
Hi @j3r3m1 @bweeding 
Actually we (@bbrangeo and I) have been using URock to run simulations linked to a weather file. Currently we run simulations every 10° at 1m/s and then estimate the distribution in the urban area at each time step of the weather file using extrapolation. We are interested in further progress.
Best regards,

**Reply by bweeding** [reply]
Hi [simonmarti1992](https://github.com/simonmarti1992), sounds interesting! 

Do you mean you are averaging between directions, and then performing some kind of linear scaling for speed? 

Will keep everyone updated as I go.
Ben
